woocommerce/woocommerce-android
# Change range id type

This PR depends on https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2863

Closes: https://github.com/woocommerce/woocommerce-android/issues/9916
Closes: https://github.com/woocommerce/woocommerce-android/issues/9734

### Why
As part of the Reliability project, we added the `RANGE_ID` field to retrieve the cached revenue stats from the DB quickly. Aiming to have a more readable and performant `ID` we decided to define this field as an Integer. To generate this field's value, we used the selected time period name, the period start date, and the period end date values. We joined all these values as a string key and used the `hashCode()` to generate the integer key. The idea was that the hashCode generated should be unique, but after a close look at the `hashCode()` function, we realized that collisions are more common than we thought.

[Here](https://play.kotlinlang.org/#eyJ2ZXJzaW9uIjoiMS42LjIxIiwicGxhdGZvcm0iOiJqYXZhIiwiYXJncyI6IiIsIm5vbmVNYXJrZXJzIjp0cnVlLCJ0aGVtZSI6ImlkZWEiLCJjb2RlIjoiLyoqXG4gKiBZb3UgY2FuIGVkaXQsIHJ1biwgYW5kIHNoYXJlIHRoaXMgY29kZS5cbiAqIHBsYXkua290bGlubGFuZy5vcmdcbiAqL1xuZnVuIG1haW4oKSB7XG4gICAgdmFsIGEgPSBcIkZCXCIuaGFzaENvZGUoKVxuICAgIHZhbCBiID0gXCJFYVwiLmhhc2hDb2RlKClcbiAgICBcbiAgICBwcmludGxuKCBcIiRhIHNob3VsZCBiZSBkaWZmZXJlbnQgdGhhbiAkYlwiKVxufSJ9), you can check two different strings with the same hashCode

### Description
This PR changes the `RANGE_ID` from Int -> String. Changing the RANGE_ID from Int to String should be enough to prevent the `RANGE_ID` collisions.

### Testing
TC1
1. Run  the app with the code on `trunk`
2. Open the MyStore stats and change through different tabs
3. Open the Analytics Hub and check different stats too 
4. Change to this branch and run the code
5. Check that the app doesn't crash and the migration runs smoothly

TC2
1. Compare that the stats on the app match the stats on the web (The app behavior is the same; nothing is affected by the new change)

### Images/gif
The Range Id is now a String key

![image](https://github.com/woocommerce/woocommerce-android/assets/18119390/98c314fa-5369-4601-9730-cd85563c4b16)
